### PR TITLE
feat(tags): Add tag management support

### PR DESCRIPTION
**Pull request by JakeMakesStuff**

### CODE REVIEW

File: core/tags.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+type tagsResponseBody struct {
+	Pagination *katapult.Pagination `json:"pagination,omitempty"`
+	Tags       []*Tag               `json:"tags,omitempty"`
+	Tag        *Tag                 `json:"tag,omitempty"`
+}
+
+func (s *TagsClient) List(
+	ctx context.Context,
+	ref OrganizationRef,
+	opts *ListOptions,
```
Comment (jimeh) [inline core/tags.go:39]: Actually, to make this more annoying, the tags list endpoint does NOT support pagination... LOL 🤦‍♂️

https://developers.katapult.io/docs/api/latest/core/groups/tags/

Though, I'd say we should probably consider making it paginated. In theory I would say all list endpoints in the API should support pagination, possibly with exceptions for specific global resources which will only contain a handful of items.

So let's discuss about the way forward here before removing the pagination stuff as it's already here :)

File: core/tags_test.go [hunk 2]
```diff
@@ -0,0 +1,615 @@
...
+			obj:  &TagArguments{},
+		},
+		{
+			name: "name",
+			obj:  &TagArguments{Name: "tag_name"},
+		},
+		{
+			name: "color",
+			obj:  &TagArguments{Color: "#0d1d1f"},
+		},
```
Comment (jimeh) [inline core/tags_test.go:126]: Nitpick: I don't think we need these test cases, as between empty and full, we're ensuring the `omitempty` (or not) behavior of each field, along with the field name when it is present.

**Review by jimeh — CHANGES_REQUESTED** [review]

File: core/tags.go (hunk 1 above)
Comment (strideynet) [reply core/tags.go:39]: ^ is this blocked on this ?

Comment (strideynet) [reply core/tags.go:39]: We are leaving the pagination stuff in.

**Review by strideynet — APPROVED** [review]
